src/services/auth.service.ts

```diff
@@ -1,13 +1,68 @@
-import { generateToken, validateToken } from "../config/jwt/tokens";
-
+import { generateToken } from "../config/jwt/tokens";
+import User from "../models/User";
+import {
+  LoginRequestBody,
+  UserRequestBody,
+} from "../controllers/auth.controller";
 class AuthService {
-  static register(user: { name: string }) {
-    const token = generateToken(user);
-    return token;
+  static async register(userBody: UserRequestBody) {
+    try {
+      const newUser = await new User(userBody).save();
+
+      if (!newUser) {
+        throw new Error("Failed to create new user");
+      }
+      const userfiltered = {
+        name: newUser.name,
+        lastname: newUser.lastname,
+        email: newUser.email,
+        is_admin: newUser.is_admin,
+        is_active: newUser.is_active,
+        urlphoto: newUser.urlphoto,
+        pendingPackages: newUser?.pendingPackages,
+        currentPackage: newUser?.currentPackage,
+        historyPackages: newUser?.historyPackages,
+      };
+      const token = generateToken(newUser._id);
+      if (!token) {
+        throw new Error("Failed to generate token");
+      }
+
+      return { token, userfiltered };
+    } catch (error) {
+      throw new Error("Registration failed");
+    }
```

Comment:
En el service en este caso te conviene directamente no catchear los errores y que los maneje directamente el controller.

```suggestion
  static async register(userBody: UserRequestBody) {
      const newUser = await new User(userBody).save();

      if (!newUser) {
        throw new Error("Failed to create new user");
      }
      const userfiltered = {
        name: newUser.name,
        lastname: newUser.lastname,
        email: newUser.email,
        is_admin: newUser.is_admin,
        is_active: newUser.is_active,
        urlphoto: newUser.urlphoto,
        pendingPackages: newUser?.pendingPackages,
        currentPackage: newUser?.currentPackage,
        historyPackages: newUser?.historyPackages,
      };
      const token = generateToken(newUser._id);
      if (!token) {
        throw new Error("Failed to generate token");
      }

      return { token, userfiltered };
    }
```

Fijate que este throw nunca se va a mantener al tener el `catch`.

Sino que siempre va a tirar este error por el catch
```
    } catch (error) {
      throw new Error("Registration failed");
    }
  ```